### ISSUE TITLE
All configuration page is unresponsive

### ISSUE THREAD

**Issue by tqvarnst** [quarkusio/quarkusio.github.io#1796]
**Describe the bug**
When I go to the all configuration options page https://quarkus.io/guides/all-config it takes a really long time for the search field to become available and Chrome says the page is unresponsive. See the attached video.

https://github.com/quarkusio/quarkusio.github.io/assets/1204115/e276f548-5237-4b80-bf3b-e3a1202863c1

**Which pages**
https://quarkus.io/guides/all-config

I'm on Mac OS using Version 116.0.5845.187 (Official Build) (x86_64) of Chrome

**Comment by tqvarnst** [comment]
https://github.com/quarkusio/quarkusio.github.io/assets/1204115/0630831c-536a-4868-864d-c7184ecb9d4c

**Comment by maxandersen** [comment]
I can confirm Chrome has a hard time on https://quarkus.io/guides/all-config. 

I had same issue in past but hadn't noticed it for a while because I use firefox and here I can the days type in text in search almost instantly (maybe 1-3s delay).

@FroMage or @ia3andy didn't you do some updates recently to this page to fix something similar to this or do i remember wrong?

**Comment by FroMage** [comment]
Not recently no, years ago I added support for shadow DOM to make filtering much faster. I guess either the option count increased.

**Comment by tqvarnst** [comment]
@FroMage, I can do it if you can send details of what profiling tool you want me to use and how to use it. 😉

**Comment by FroMage** [comment]
Ctrl-Shit-I (to open the inspector), Go to `Performance` tab, and click Record (or Ctrl-E):

This is what happens when I type `panache` in the search field:

![Screenshot from 2023-09-28 11-52-55](https://github.com/quarkusio/quarkusio.github.io/assets/179265/b9a1270e-d1bc-4c09-8dca-aa7fc230ea66)

100ms search and update, not bad.

You can also start recording and reload, there's an option for that (or Ctrl-Shift-E):

![Screenshot from 2023-09-28 11-58-00](https://github.com/quarkusio/quarkusio.github.io/assets/179265/d5128772-6692-47e7-8b3d-c88eab5567c6)

Next week I'll have access to a Mac, I can try it then. Not sure a remote debug session will help us much :(

**Comment by michalvavrik** [comment]
I don't think this can improve as long as all configuration properties that fit the search term are rendered. IMHO this can only be fixed with lazy loading on scroll down. Now that there is actual back end used for searching, can it be also used to serve configuration properties that match given term (paginated sorted result). 

@yrodiere WDYT? I have actually wanted to do that for a long time (both FE and BE part), but I somehow expected that this is priority and someone already has plan to fix it. @insectengine @gsmet do you happen to know what is current state?

I'll wait, but if there is consent that proposed solution would do, I'm happy to have a look. Thanks

**Comment by FroMage** [comment]
I frankly doubt that we'll get remote search results faster than 100ms. Not to mention that the returned results page size will be gigantic for the first letters typed.

**Comment by michalvavrik** [comment]
> Not to mention that the returned results page size will be gigantic for the first letters typed.

I mentioned lazy loading, what I am suggesting is to serve content in chunks to front end that loads it on demand (scroll).

**Comment by yrodiere** [comment]
> [@yrodiere](https://github.com/yrodiere) WDYT? I have actually wanted to do that for a long time (both FE and BE part), but I somehow expected that this was a priority and someone already had already planned to fix it. [@insectengine](https://github.com/insectengine) [@gsmet](https://github.com/gsmet) do you happen to know what is current state?

I think you're looking for https://github.com/quarkusio/search.quarkus.io/issues/313. If you want to make suggestions regarding a remote search endpoint, that would be the place.

I think the idea would be to have some yaml/json file with the data about configuration properties available somewhere, and to have search.quarkus.io index that, so that it can expose a search service. You could then change the page on quarkus.io to _not_ display all properties by default, but just offer a search box and display the ~50 first matches by default, then lazy-load the next matches (we already have the frontend code for something similar). And you'd be done.

I cannot personally make this a priority, but we gladly accept PRs.

**Comment by michalvavrik** [comment]
Thank you @yrodiere , I'll definitely look into it. I'll start by the end of the week, if anyone has different suggestions / preferences, there is time.

**Comment by marko-bekhta** [comment]
One thing I'd add is that if we end up changing the search on that page, should we also consider changing how that page looks? 
My thinking here is that if we show just a "first page" of config properties (even if sorted by an extension) we'll lose the current table-lake grouped structure. Hence, maybe we should add some sort of facet-like filters so that users can still easily find all the properties for a particular extension..

> feel free to ping @marko-bekhta or me, we can help you get started with the codebase.

+1 😃

**Comment by insectengine** [comment]
Looking at the web stats for December, this page is the 12th most popular (of ~1800) so making it more user friendly would be a benefit for the users.

Would definitely be worth a conversation on how to organize the content in order to display it better display it besides the tables.

I'd say before running off to code this, we should determine the final form of the page/content.

**Comment by insectengine** [comment]
Also, isn't this edited via asciidoc?  Seems like wholesale changes to this would impact the editability and potentially backwards compatibility to previous versions of the guides.

**Comment by michalvavrik** [comment]
sure @gsmet , I need to start working on it before elaborating more. For me, when I use the all config page, my user experience is still bad.

> Also, isn't this edited via asciidoc? Seems like wholesale changes to this would impact the editability and potentially backwards compatibility to previous versions of the guides.

You decide what you generate with ASCIIDOC, if I decide to additionally to generate json object or something, there is no reason why it can't be compatible with what other versions do. What I take from that @insectengine is that I should keep in mind backwards compatibility. Thanks, good point.

**Comment by gsmet** [comment]
Yeah so the thing is that some context is kept by the previous rows. If you just get the JSON properties from a search engine, you might miss some context (like the section or the extension).

So you need to make sure you push the sections + the extension when indexing.

If you experiment with the current search, you will see we try hard keeping the extension + section context (this is recent, thus why I underline it).
